plugin.video.anime/main.py

```diff
@@ -126,63 +136,45 @@ def ListowanieAnime():
 
 def ListowaniOdcinkow():
     url = params['url']
-
-    try:
-        iconimage = params['iconimage']
-    except:
-        iconimage = ''
+    iconimage = params.get('iconimage', '')
 
     r = client.request(url)
-
-    linki = []
-    nazwy = []
-
     result = client.parseDOM(r, 'table', attrs={'class': 'table table-bordered table-striped table-hover episodes'})
-    test = client.parseDOM(result, 'tr')
-    test = [x for x in test if str(x).startswith("<td class=\"text")]
-    for item in test:
-        link = client.parseDOM(item, 'a', ret='href')
-        nazwa = client.parseDOM(item, 'td', attrs={'class': 'episode-title'})
-        odcinek = client.parseDOM(item, 'strong')[0]
-        if nazwa and link:
-            linki.append(link[0])
-            nazwy.append("Odcinek " + odcinek + " " + nazwa[0])
+    result = client.parseDOM(result, 'tbody')
+    for tr in client.parseDOM(result, 'tr'):
+        link = client.parseDOM(tr, 'a', ret='href')[0]
+        nazwa = client.parseDOM(tr, 'td', attrs={'class': 'episode-title'})[0]
+        odcinek = client.parseDOM(tr, 'strong')[0]
+        lang = client.parseDOM(tr, 'span', ret='class')[0].split()[-1]
 
-    counter = 0
-    for link in linki:
-        linki[counter] = 'http://animezone.pl' + str(link).replace("..", "")
-        addon.addDir(str(nazwy[counter]), linki[counter], mode=5, icon=iconimage)
-        counter += 1
+        nazwa = '{ep:02d}. {title} ({lang})'.format(title=nazwa, ep=int(odcinek), lang=lang)
+        addon.addDir(nazwa, urljoin(url, link), mode=5, icon=iconimage)
 
 
 def WysiwetlanieLinkow():
-    global url
-    global nazwa
-
     url = params['url']
-    r = s.get(url).content
-
-    nazwa = params['name']
-    results = client.parseDOM(r, 'table', attrs={'class': 'table table-bordered table-striped table-hover episode'})
-    results = client.parseDOM(results, 'td', attrs={'class': 'text-center'})
-    counter = range(-1, len(results) - 1)
-    for zipped in zip(counter, results):
-        results.append(work(zipped))
-    results = [x for x in results if type(x) == tuple]
-    for result in results:
-        addon.addLink(result[0], result[1], mode=6)
-
-
-def work(zipped):
-    counter = zipped[0]
-    result = zipped[1]
-    counter += 1
-
-    try:
-        r = re.findall("""data-.*="([0-9].*)">""", result)[0]
-    except:
-        return 0
-
+    name = params['name']
+
+    with requests.session() as sess:
+        r = sess.get(url).text
+        result = client.parseDOM(r, 'table', attrs={'class': 'table table-bordered table-striped table-hover episode'})
+        result = client.parseDOM(result, 'tbody')
+        for tr in client.parseDOM(result, 'tr'):
+            r = re.search(r'(?is)[^<]*(?:<tr[^]>*>[^<]*)?<td[^>]*>(?P<name>[^<]*)<.*?'
```

Comment:
Sam się skomentuję. ParseDOM jest na tyle toporna, że łatwiej mi było użyć własnego drobnego, trzylinijkowego wyrażenia regularnego :-)
Z pdom.select, byłoby to dużo prostsze, ale to jeszcze moment.